### PR TITLE
Replace if judge with MAX() function in scale less than 1

### PR DESCRIPTION
**Pull request by kinarob**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description

It does not change any logic in the code, Just to make the code  in a pithy style！

### CODE REVIEW

File: SDWebImage/SDWebImageDefine.m [hunk 1]
```diff
@@ -39,14 +39,14 @@ inline CGFloat SDImageScaleFactorForKey(NSString * _Nullable key) {
                 NSString *fileScale = [NSString stringWithFormat:@"@%@x.", scaleFactor];
                 if ([key containsString:fileScale]) {
                     scale = scaleFactor.doubleValue;
-                    return scale;
+                    return MAX(scale, 1);
```
Comment (dreampiggy) [inline SDWebImage/SDWebImageDefine.m]: Actually, this `scale` is from internal const array `SDImageScaleFactors()`, which can not be less than 1...

Comment (kinarob) [reply SDWebImage/SDWebImageDefine.m]: Oh,  When enter `SDImageScaleFactorForKey()`, It's not necessary to judge whether it's less than 1 ？

File: SDWebImage/SDImageAPNGCoder.m [hunk 2]
```diff
@@ -83,10 +83,7 @@ - (UIImage *)decodedImageWithData:(NSData *)data options:(nullable SDImageCoderO
     CGFloat scale = 1;
     NSNumber *scaleFactor = options[SDImageCoderDecodeScaleFactor];
     if (scaleFactor != nil) {
```
Comment (zhongwuzw) [inline SDWebImage/SDImageAPNGCoder.m]: @kinarob I think you can even remove all these nil check, just one line `scale = MAX([scaleFactor doubleValue], 1);`. 😄

Comment (kinarob) [reply SDWebImage/SDImageAPNGCoder.m]: Good point！

**Pull request merged**